MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView.swift

```diff
@@ -23,13 +24,18 @@ class SubcategoryView: UIView {
         fatalError("init(coder:) has not been implemented")
     }
     
-    func populate(with subcategories: [LocalSubcategory]) {
+    func setDelegate(delegate: CategoryCellDelegate) {
+        self.delegate = delegate
+    }
+    
+    func populate(with subcategories: [LocalSubcategory]?) {
+        guard let subcategories = subcategories else { return }
+        
         self.subcategories = subcategories
-                
-        if subcategoriesStack.arrangedSubviews.isEmpty {
-            selectedSubcategory = self.subcategories?[0].rawValue
-            addButtons()
-        }
+        selectedSubcategory = self.subcategories?[0].rawValue
+        addButtons()
```

Comment:
Buttons now refresh every time populate function is called.
In function `addButton()` all subviews of `UIStackView` are deleted and afterwards are new buttons added.